pkg/virtualgarden/kube_api_server.go

```diff
@@ -39,7 +39,27 @@ func (o *operation) DeployKubeAPIServer(ctx context.Context) error {
 		return err
 	}
 
-	_, err = o.deployKubeAPIServerSecrets(ctx, checksums)
+	basicAuthPw, err := o.deployKubeAPIServerSecrets(ctx, checksums)
+	if err != nil {
+		return err
+	}
+
+	err = o.deployKubeAPIServerConfigMaps(ctx, checksums)
+	if err != nil {
+		return err
+	}
+
+	err = o.deployKubeAPIServerPodAutoscaling(ctx)
+	if err != nil {
+		return err
+	}
+
+	err = o.deployMisc(ctx)
+	if err != nil {
+		return err
+	}
+
+	err = o.deployDeployments(ctx, checksums, basicAuthPw)
```

Comment:
I see no problem to separate the last 4 steps to make the flow more explicit but I'm not sure if we should separate and parallelise the small steps like computeKubeAPIServerLoadBalancer, deployKubeAPIServerSecrets etc. On the one hand we will not save much time because all steps are very fast except waiting for the load balancer. On the other hand, we make the logic more complex without a real benefit, e.g. the load balancer step must be executed before the computation of the certificates, we have more confusing log entries, need synchronised data structures for the checksums and the flow graph becomes more complex.

Reply:
Ok